src/Common/src/Interop/Unix/System.Native/Interop.ForkAndExecProcess.cs

```diff
@@ -31,17 +31,11 @@ internal static partial class Sys
                     out lpChildPid, out stdinFd, out stdoutFd, out stderrFd);
                 if (result != 0)
                 {
-                    // Normally we'd simply make this method return the result of the native
-                    // call and allow the caller to use GetLastWin32Error.  However, we need
-                    // to free the native arrays after calling the function, and doing so
-                    // stomps on the runtime's captured last error.  So we need to access the
-                    // error here, and without SetLastWin32Error available, we can't propagate
-                    // the error to the caller via the normal GetLastWin32Error mechanism.  We could
-                    // return 0 on success or the GetLastWin32Error value on failure, but that's
-                    // technically ambiguous, in the case of a failure with a 0 errno.  Simplest
-                    // solution then is just to throw here the same exception the Process caller
-                    // would have.  This can be revisited if we ever have another call site.
-                    throw new Win32Exception();
+                    return Marshal.GetLastWin32Error();
```

Comment:
Boy this is a bad API name in hindsight. I wonder whether we should have something like `Marshal.GetLastError()` that does the same thing and recommend cross plat code move to that...

Reply:
A name that doesn't have `Win32` would be nicer indeed.